### ISSUE TITLE
Inside of FrameGenerator.Generate instead of HSL Blocks use SourceReader bytestream instead 

### ISSUE THREAD

**Issue by andreysaf** [smourier/VCamNetSample#2]
I was wondering if what I'm trying to do is technically feasible and if there's anything I should watch out for? Any samples would be greatly appreciated 🙏.

Instead of drawing HSL blocks, I want to pass a bytestream from the source reader callback of a web camera and use renderTarget to create a bitmap of the frame. Then, I want to draw on top of that frame. In simpler terms, I want to watermark the frame of a physical device and then serve it as a virtual cam.

My initial thoughts:
1. Create new class called `MediaSourceReader`.
2. Implement all required functions for the source reader.
3. Inside of `OnReadSample` callback make the buffer available to the `FrameGenerator`.
4. Inside of generate method, grab the buffer and draw on top.

I tried splicing VCamSample and [SourceReader Code](https://elcharolin.wordpress.com/2017/08/28/webcam-capture-with-the-media-foundation-sdk/) but ran into compilation issues with VCamSample:
![image](https://github.com/user-attachments/assets/e5baf542-855e-4a01-8128-e1ea7e84d945)

I'm not a C++ developer, so your library is a lifesaver.

**Comment by smourier** [comment]
Hi,

The original Microsoft official sample here https://github.com/microsoft/Windows-Camera/tree/master/Samples/VirtualCamera is twofold, one similar to mine named "SimpleMediaSource" and another "AugmentedMediaSource " that "will wrap an existing camera on the system" so I guess you'd would want to take a look at it. Unfortunately, I find these samples quite complicated and the augmented one even worse (that's actually why I create mine...) and ... they are not .NET!

Otherwise, w/o even using the Microsoft sample as a start, I suppose a simple approach like the one you describe seems feasible (maybe not optimal, but what's optimal depends on your context) but I don't have any other code to share, this would require some amount of time to develop I guess.

**Comment by andreysaf** [comment]
Thank you! I will keep trying and go from there.
